pkg/sdk/testint/procedures_integration_test.go

```diff
@@ -0,0 +1,541 @@
+			WithAs(sdk.String(as))
+		err := client.Procedures.CreateProcedureForPython(ctx, request)
+		require.NoError(t, err)
+		t.Cleanup(cleanupProcedureHandle(id, []string{"VARCHAR", "VARCHAR"}))
+
+		procedures, err := client.Procedures.Show(ctx, sdk.NewShowProcedureRequest())
+		require.NoError(t, err)
+		require.GreaterOrEqual(t, len(procedures), 1)
+	})
+
+	t.Run("create procedure for SQL: returns result data type", func(t *testing.T) {
+		name := "output_message"
+		id := sdk.NewSchemaObjectIdentifier(databaseTest.Name, schemaTest.Name, name)
+
+		as := `
+	BEGIN
+		RETURN message;
+	END;`
+
+		resultDataType := sdk.NewProcedureReturnsResultDataTypeRequest().WithResultDataType("VARCHAR")
+		procedureReturns := sdk.NewProcedureReturns3Request().WithResultDataType(resultDataType).WithNotNull(sdk.Bool(true))
+		procedureArgument := sdk.NewProcedureArgumentRequest().WithArgName("message").WithArgDataType("VARCHAR")
+		request := sdk.NewCreateProcedureForSQLProcedureRequest(id).
+			WithOrReplace(sdk.Bool(true)).
+			WithArguments([]sdk.ProcedureArgumentRequest{*procedureArgument}).
+			WithReturns(procedureReturns).
+			WithAs(as)
+		err := client.Procedures.CreateProcedureForSQL(ctx, request)
+		require.NoError(t, err)
+		t.Cleanup(cleanupProcedureHandle(id, []string{"VARCHAR"}))
+
+		procedures, err := client.Procedures.Show(ctx, sdk.NewShowProcedureRequest())
+		require.NoError(t, err)
+		require.GreaterOrEqual(t, len(procedures), 1)
+	})
+
+	t.Run("create procedure for SQL: returns table", func(t *testing.T) {
+		name := "find_invoice_by_id"
+		id := sdk.NewSchemaObjectIdentifier(databaseTest.Name, schemaTest.Name, name)
+
+		as := `
+	DECLARE
+		res RESULTSET DEFAULT (SELECT * FROM invoices WHERE id = :id);
+	BEGIN
+		RETURN TABLE(res);
+	END;`
+		column1 := sdk.NewProcedureColumnRequest().WithColumnName("id").WithColumnDataType("INTEGER")
+		column2 := sdk.NewProcedureColumnRequest().WithColumnName("price").WithColumnDataType("NUMBER(12,2)")
+		procedureReturnsTable := sdk.NewProcedureReturnsTableRequest().WithColumns([]sdk.ProcedureColumnRequest{*column1, *column2})
+		procedureReturns := sdk.NewProcedureReturns3Request().WithTable(procedureReturnsTable)
+		procedureArgument := sdk.NewProcedureArgumentRequest().WithArgName("id").WithArgDataType("VARCHAR")
+		request := sdk.NewCreateProcedureForSQLProcedureRequest(id).
+			WithOrReplace(sdk.Bool(true)).
+			WithArguments([]sdk.ProcedureArgumentRequest{*procedureArgument}).
+			WithReturns(procedureReturns).
+			WithAs(as)
+		err := client.Procedures.CreateProcedureForSQL(ctx, request)
+		require.NoError(t, err)
+		t.Cleanup(cleanupProcedureHandle(id, []string{"VARCHAR"}))
+
+		procedures, err := client.Procedures.Show(ctx, sdk.NewShowProcedureRequest())
+		require.NoError(t, err)
+		require.GreaterOrEqual(t, len(procedures), 1)
+	})
+}
+
+func TestInt_AlterAndShowProcedures(t *testing.T) {
+	client := testClient(t)
+	ctx := context.Background()
```

Comment:
we also do have a ctx that could be reused

Reply:
fixed